apps/vaults/components/details/actions/QuickActionsButtons.tsx

```diff
@@ -0,0 +1,147 @@
+import React, {useCallback, useEffect, useMemo, useState} from 'react';
+import {ethers} from 'ethers';
+import useSWR from 'swr';
+import {useActionFlow} from '@vaults/contexts/useActionFlow';
+import {Solver, useSolver} from '@vaults/contexts/useSolver';
+import {getEthZapperContract} from '@vaults/utils';
+import {Button} from '@yearn-finance/web-lib/components/Button';
+import {useSettings} from '@yearn-finance/web-lib/contexts/useSettings';
+import {useWeb3} from '@yearn-finance/web-lib/contexts/useWeb3';
+import {useChainID} from '@yearn-finance/web-lib/hooks/useChainID';
+import {toAddress} from '@yearn-finance/web-lib/utils/address';
+import {ETH_TOKEN_ADDRESS} from '@yearn-finance/web-lib/utils/constants';
+import {defaultTxStatus} from '@yearn-finance/web-lib/utils/web3/transaction';
+import {useWallet} from '@common/contexts/useWallet';
+import {useAllowanceFetcher} from '@common/hooks/useAllowanceFetcher';
+import {DefaultTNormalizedBN} from '@common/utils';
+
+import type {ReactElement} from 'react';
+import type {SWRResponse} from 'swr';
+import type {TAddress} from '@yearn-finance/web-lib/utils/address';
+import type {TNormalizedBN} from '@common/types/types';
+
+
+function	shouldUseVanillaAllowance(solver: Solver, isInputTokenEth: boolean): boolean {
+	return ([Solver.VANILLA, Solver.CHAIN_COIN, Solver.PARTNER_CONTRACT].includes(solver) && !isInputTokenEth);
+}
```

Comment:
```suggestion
function	shouldUseVanillaAllowance({solver, isInputTokenEth}: {solver: Solver, isInputTokenEth: boolean}): boolean {
	return ([Solver.VANILLA, Solver.CHAIN_COIN, Solver.PARTNER_CONTRACT].includes(solver) && !isInputTokenEth);
}
```
and then
```typescript
	const withVanillaAllowance = shouldUseVanillaAllowance({
		solver: currentSolver,
		isInputTokenEth: selectedOptionFrom?.value === ETH_TOKEN_ADDRESS
	});
```

So it's a bit clearer what `selectedOptionFrom?.value === ETH_TOKEN_ADDRESS` is for